src/Client.php

```diff
@@ -343,57 +343,57 @@ private function applyOptions(RequestInterface $request, array &$options): Reque
             'set_headers' => [],
         ];
 
-        if (isset($options['headers'])) {
-            $modify['set_headers'] = $options['headers'];
-            unset($options['headers']);
+        if (isset($options[RequestOptions::HEADERS])) {
+            $modify['set_headers'] = $options[RequestOptions::HEADERS];
+            unset($options[RequestOptions::HEADERS]);
         }
 
-        if (isset($options['form_params'])) {
-            if (isset($options['multipart'])) {
+        if (isset($options[RequestOptions::FORM_PARAMS])) {
+            if (isset($options[RequestOptions::MULTIPART])) {
                 throw new InvalidArgumentException('You cannot use '
                     . 'form_params and multipart at the same time. Use the '
                     . 'form_params option if you want to send application/'
                     . 'x-www-form-urlencoded requests, and the multipart '
                     . 'option to send multipart/form-data requests.');
             }
-            $options['body'] = \http_build_query($options['form_params'], '', '&');
-            unset($options['form_params']);
+            $options[RequestOptions::BODY] = \http_build_query($options[RequestOptions::FORM_PARAMS], '', '&');
+            unset($options[RequestOptions::FORM_PARAMS]);
             // Ensure that we don't have the header in different case and set the new value.
             $options['_conditional'] = Psr7\Utils::caselessRemove(['Content-Type'], $options['_conditional']);
             $options['_conditional']['Content-Type'] = 'application/x-www-form-urlencoded';
         }
 
-        if (isset($options['multipart'])) {
-            $options['body'] = new Psr7\MultipartStream($options['multipart']);
+        if (isset($options[RequestOptions::MULTIPART])) {
+            $options[RequestOptions::BODY] = new Psr7\MultipartStream($options[RequestOptions::MULTIPART]);
             unset($options['multipart']);
```

Comment:
This occurrence of `$options['multipart']` has been intentionally left unchanged.

If replaced with `$options[RequestOptions::MULTIPART]` (which is equivalent as `RequestOptions::MULTIPART` === `multipart`), psalm fails with the error:

```
::error file=src/Client.php,line=450,col=22::Cannot iterate over string
```

I can't see any meaningful reason why this failure should be happening.